sap_swpm: Option to enable SWPM observer mode

rob0d opened the pull request:
#709

Wabri commented on roles/sap_swpm/tasks/pre_install.yml:
```diff
@@ -15,13 +15,19 @@
     # IF SWPM is running a HA installation, Ansible Variable sap_swpm_swpm_command_virtual_hostname is set and contains "SAPINST_USE_HOSTNAME={{ sap_swpm_virtual_hostname }} IS_HOST_LOCAL_USING_STRING_COMPARE=true"
     # If SWPM is running a MP Stack XML installation, Ansible Variable sap_swpm_swpm_command_mp_stack is set and contains "SAPINST_STACK_XML={{ sap_swpm_mp_stack_path }} + '/' (if needed) + {{ sap_swpm_mp_stack_file_name }}"
-    sap_swpm_swpm_command_extra_args: "SAPINST_SKIP_DIALOGS=true SAPINST_START_GUISERVER=false {{ sap_swpm_swpm_command_virtual_hostname }} {{ sap_swpm_swpm_command_mp_stack }}"
+    #RDTEST
```
I would also remove the two lines commented below at line 29 and 30

rob0d replied:
Sorry, missed those. Removed now.

sean-freeman reviewed (changes requested):
@rob0d I'm good to merge this PR, it's non-destructive and can work before the overhaul of the sap_swpm code structure in the coming weeks.

However, in defaults the `sap_swpm_swpm_remote_access_user` should be uncommented and I would recommend it is by default set to `root`. If an end user wants to enable this feature and observe using a different OS User, then it can be changed of course. All SWPM executions are otherwise `root` so I think this default is OK.

berndfinger commented:
Hi @rob0d, we are working on finalizing a new version, and your PR can be part of it. Can you please respond to the suggestion from @sean-freeman?

rob0d commented:
@sean-freeman  @berndfinger  apologies, a bit snowed under.
I would prefer not to do that as it will change the role's default behaviour.
Currently when SWPM is executed `SAPINST_REMOTE_ACCESS_USER=` and `SAPINST_REMOTE_ACCESS_USER_IS_TRUSTED=true` are not set.  This makes SWPM use root (unix) or Administrator(windows) automatically.
By setting the default to root, these two parameters will be set even thought it's not required as they basically emulate the default behaviour. I have never tried to use `SAPINST_REMOTE_ACCESS_USER=root` and I don't know how  will SWPM react. It _should_ behave the same way, but I am not sure and I won't be able to test it for at least 4 weeks. 

So setting the default to root, will change the current default behaviour of the role and I am not sure what impact it will have.

berndfinger commented:
> So setting the default to root, will change the current default behaviour of the role and I am not sure what impact it will have.

@rob0d I think in this case, we better not merge this PR into `1.4.1` but continue this discussion afterwards, for including it into `1.4.2`, together with more changes for role `sap_swpm`.

sean-freeman commented:
> ... in defaults the `sap_swpm_swpm_remote_access_user` should be uncommented and I would recommend it is by default set to `root`. If an end user wants to enable this feature and observe using a different OS User, then it can be changed of course. All SWPM executions are otherwise `root` so I think this default is OK.

Still awaiting this change to be made in defaults, otherwise PR is blocked

rob0d commented:
Updated the default to be an empty variable and added code to handle this in a way that it doesn't change SWPM parameters and behaviour.
Unless this is enabled by setting `sap_swpm_swpm_observer_mode` to true, the behaviour is the same as before, so this is a non breaking change.
If `sap_swpm_swpm_remote_access_user` is not explicitly set and `sap_swpm_swpm_observer_mode = true`, it will execute SWPM with standard parameters.
If `sap_swpm_swpm_remote_access_user` is set,  it will add appropriate SWPM command line parameters to use the requested user.

sean-freeman reviewed (approved):
lgtm 👍

berndfinger reviewed (approved):
LGTM
